### ISSUE TITLE
Git diff/status doesn't have color but Git diff --staged does

### ISSUE THREAD

**Issue by thekenu** [tpope/vim-fugitive#1552]
<del> 
Disclaimer: I tried to isolate the problem by removing all other plugins and making sure no weird scripts are loaded in `:scriptnames`. I am using gnome-terminal on Ubuntu 18.04

The temporary buffer opened by `:Git diff` and `:Git status` is _not_ colored, 
![](https://user-images.githubusercontent.com/16970019/84828387-d638cf00-afda-11ea-8c3f-b8af41cfac2e.png)

but `:Git diff --staged` is just fine:

![](https://user-images.githubusercontent.com/16970019/84828437-e781db80-afda-11ea-8dc4-436f88769085.png)

I'm a bit confused because I thought `:Git diff --staged` would be the same as `:Git diff`, just with extra flags. I also did some testing with commands that start with `G` instead of `Git`, such as `:Gstatus` and `:Glog`. And the colors show up just fine.

I do _not_ think `vim-fugitive` is broken, bu I would like to get some help on how I can further narrow down the problem to figure out what's wrong with my setup.
</del>

**Comment by thekenu** [comment]
I found a way to eproduce the problem. When I did `git diff` the temp buffer starts with `warning: CRLF will be replaced ...` and that seems to mess up the syntax highlighting. If my staging area doesn't have any `warning: CRLF will be replaced ...`, then the syntax highlighting works again!

`warning: CRLF will be replaced ...` would never show up in `git diff --staged`, which explains the behaviour I was observing.
